lib/symboldatabase.cpp

```diff
@@ -3585,8 +3585,10 @@ std::string Type::name() const
         start = start->tokAt(1);
     else if (start->str() == "class")
         start = start->tokAt(1);
-    else if (!start->isName())
-        return emptyString;
+    else if (!start->isName()) {
+        static const std::string s_empty_string;
```

Comment:
Has anyone ever benchmarked this? I wonder if the static object can provide any benefits, since it is returned immediately, and RVO should just construct the object at the target location anyway.

Reply:
Oh. I expected this to be a const reference. Will back out.

> Has anyone ever benchmarked this?

I have at one point with it being dropped completely and it caused a sizable performance impact. That's why I am doing the removal in steps so it is easier to pinpoint potential regressions. Will profile it again after I get rid of the cold stuff.

The whole thing might only be necessary with older C++ standards (pre-C++17?).

Reply:
With current GCC, the static version is even slower: https://quick-bench.com/q/goY88i2idDwpvg2VMh9bquG1Tds
No difference with clang. Have not checked older standards.

Reply:
Interesting. Good to know ahead of doing the profiling for this.

FYI some slightly related things:
https://github.com/llvm/llvm-project/issues/58002
https://github.com/llvm/llvm-project/issues/60165

Reply:
Done.

Reply:
Done.